### PR TITLE
More statistics validation

### PR DESCRIPTION
**Pull request by bramkragten**
Needs https://github.com/home-assistant/core/pull/56892

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/developer-tools/statistics/developer-tools-statistics.ts [hunk 1]
```diff
@@ -119,48 +127,115 @@ class HaPanelDevStatistics extends LitElement {
...
+      case "unsupported_unit_state":
+        showAlertDialog(this, {
+          title: "Unsupported unit",
+          text: html`The unit of your entity is not a supported unit for the
+            device class of the entity, ${issue.data.device_class}.
+            <br />Statistics can not be generated until this entity has a
+            supported unit.<br /><br />If this unit was provided by an
+            integration, this is a bug. Please report an issue.<br /><br />If
+            you have set this unit yourself, and want to have statistics
+            generated, make sure the unit matched the device class. The
```
Comment (spacegaier) [inline src/panels/developer-tools/statistics/developer-tools-statistics.ts:214]:
```suggestion
            generated, make sure the unit matches the device class. The
```

File: src/panels/developer-tools/statistics/developer-tools-statistics.ts [hunk 2]
```diff
@@ -119,48 +127,115 @@ class HaPanelDevStatistics extends LitElement {
...
+    );
+    const issue = issues[0];
+    switch (issue.type) {
+      case "entity_not_recorded":
+        showAlertDialog(this, {
+          title: "Entity not recorded",
+          text: html`State changes of this entity are not recorded, therefore,
+            we can not track long term statistics for it. <br /><br />You
+            probably excluded this entity, or have just included some
+            entities.<br /><br />See the
```
Comment (spacegaier) [inline src/panels/developer-tools/statistics/developer-tools-statistics.ts:167]:
```suggestion
            other entities.<br /><br />See the
```

Comment (bramkragten) [reply src/panels/developer-tools/statistics/developer-tools-statistics.ts:167]: I don't think that makes it clearer?

**Review by balloob — APPROVED** [review]

File: src/panels/developer-tools/statistics/developer-tools-statistics.ts (hunk 2 above)
Comment (balloob) [reply src/panels/developer-tools/statistics/developer-tools-statistics.ts:167]: I agree, it's fine as is.

**Pull request merged**